### ISSUE TITLE
Training procedure of EfficientPose

### ISSUE THREAD

**Issue by sh8** [ybkscht/EfficientPose#3]
Hi authors,

Thank you for the great work and paper!

This issue is related with [this](https://github.com/chensong1995/HybridPose/issues/32) in HybridPose.
In convention, recent methods (PoseCNN, BB8, PVNet, CDPN, and DPOD) train their networks using the LineMOD dataset and synthetic images to avoid overfitting, and then test the trained models on both the LineMOD and Occlusion LineMOD dataset.
The reason of this is If models are trained and evaluated separately, you can easily achieve the better performance on both datasets because of overfitting.
For example, HybridPose achieved 94.5% and 79.2% on the LineMOD and Occlusion LineMOD datasets respectively using the models trained separately by only real images before [1]. However, after following the convention, their score was dropped to 91.3% and 47.5% [2].

Could you please check if the same thing will happen to your network?
I'd say it's important to fairly compare the state-of-the-art methods.

Thanks,
Shun

[1] HybridPose: 6D Object Pose Estimation under Hybrid Representations, Chen Song, et al., https://arxiv.org/pdf/2001.01869v1.pdf, version 1
[2] HybridPose: 6D Object Pose Estimation under Hybrid Representations, Chen Song, et al., https://arxiv.org/pdf/2001.01869.pdf, version 4

**Comment by ybkscht** [comment]
Hi Shun,

Thanks for your comment. 
You are absolutely right that our reported results on the Occlusion dataset are not comparable with other works.
But that also wasn't our intention with this experiment. 
Instead, we wanted to verify the ability of our method to estimate the 6D poses of multiple objects with a single model. The Linemod dataset is not well designed for this case, so we trained a single model on the Occlusion dataset to show that it is possible.
We also stated in the paper that our reported Occlusion results are not comparable with the state-of-the-art and I would expect a similar drop in accuracy like HybridPose reported when using the conventional training procedure, because testing on an unkown scene is a much harder task. This is also the reason why we did not include any results from other works in the table containing our Occlusion results.

Our reported results on Linemod on the other hand should be comparable with the state-of-the-art, since we use the same dataset split as other works and the training data would remain the same because the whole Occlusion dataset is only used for testing.
It is correct that we do not use additional synthetic data but I think that's more of a disadvantage for our method.
As far as I understood, the reason for the accuracy drop of HybridPose on Linemod was not because the additional use of synthetic data but the switch from their own dataset split to the conventional split, which significantly reduced the amount of real training images from 80% to 15% (as stated [here]( https://github.com/chensong1995/HybridPose/issues/31)). This makes it of course much harder to achieve a comparable performance.
Similarly the (Linemod) test set got much bigger from 20% to 85% of the overall dataset which also makes it difficult to really compare the reported accuracys of 94.5% and 91.3%, especially when the amount of data is so small.

The only point I can see here that could possibly be negatively affected is when we check for the best ADD(-S) accuracy when evaluating within the training progress. Because the best result for Linemod and Occlusion combined is not necessarily the best for Linemod only as well.
So here could be a potential issue but I don't think that this would result in a significant drop in accuracy and the comparison with the state-of-the-art methods should still be valid in my opinion (of course only on the Linemod dataset).

Nevertheless, we are planning to use additional synthetic data and follow the convention to also publish a result on the Occlusion dataset, wich is comparable with other works (and update the Linemod results as well).
Unfortunately, we were not able to do this already in the initial version of the paper because this project and paper was developed within my master thesis and time was limited.

I hope I got your point right but let me know if I missed something.

Sincerely,
Yannick

**Comment by BlueBird29** [comment]
Hi! I was just wondering what the convention was for training/testing on the Occlusion dataset. It seems like most 6DOF models train a separate model for each class. Are the reported results in most papers on the Occlusion dataset gotten by running each of the models separately on each scene in the occlusion test set?

**Comment by ybkscht** [comment]
Hi BlueBird29,

Yes, to the best of my knowledge this is usually the case.
Like Shun already said, the convention is to use the entire Occlusion dataset only for testing. So the models are trained only on the Linemod train set (plus synthetic data) and are tested on the Linemod test set as well as the Occlusion dataset.
Since the Linemod dataset is not well suited for multi object pose estimation, usually a separate model for each object is trained.
The problem here is that there are separate scenes/sequences for each object in the Linemod dataset in which only a single object is annotated but other objects are also visible.
This makes it difficult to train a single model for all objects because the model would often be penalized unnecessarily when detecting an object (actually correctly) but since there are no annotations for this object, the provided "ground truth" at this point would be background.
This is the reason why we also trained a single model on the Occlusion dataset to see if our method is not only theoretically but also practically capable of performing 6D pose estimation for multiple objects.

Sincerely,
Yannick
